compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -793,12 +793,14 @@ impl<'interner> Monomorphizer<'interner> {
                 // Default any remaining unbound type variables.
                 // This should only happen if the variable in question is unused
                 // and within a larger generic type.
-                let default =
-                    if self.is_range_loop && matches!(kind, TypeVariableKind::IntegerOrField) {
-                        Type::default_range_loop_type()
-                    } else {
-                        kind.default_type()
-                    };
+                let default = if self.is_range_loop
+                    && (matches!(kind, TypeVariableKind::IntegerOrField)
+                        || matches!(kind, TypeVariableKind::Integer))
+                {
+                    Type::default_range_loop_type()
+                } else {
+                    kind.default_type()
+                };
```

Comment:
We should be able to remove `is_range_loop` entirely from the monomorphizer now. This check can then just be `kind.default_type()` since we'll always unify ranges with TypeVariableKind::Integer now.

Reply:
I suppose we can delay this change for a separate PR as well.